osxkeychain.go

```diff
@@ -346,12 +347,19 @@ func GetAllAccountNames(serviceName string) (accountNames []string, err error) {
 
 	defer C.CFRelease(resultsRef)
 
+	// Check that the resultsRef is an array
```

Comment:
Yeah this is a sanity thing.. It's nice to have if we add a fetch "reference" capability in the future. I added this in when I was testing other fetch methods and thought it was useful to keep in.